packages/core/src/data-access/projects.ts

```diff
@@ -0,0 +1,31 @@
+import { database } from '$core/client'
+import { Result } from '$core/lib'
+import { projects } from '$core/schema'
+import { and, eq } from 'drizzle-orm'
+
+class ProjectNotFoundError extends Error {
+  constructor() {
+    super('Project not found')
+  }
+}
+
+export async function findProject({
+  projectId,
+  workspaceId,
+}: {
+  projectId: number
+  workspaceId: number
+}) {
```

Comment:
Do we need a workspaceId to find the project?

Reply:
Yes. We only want to find the right combination of project Y that belongs to workspace X